src/scenedit/scen.main.cpp

```diff
@@ -84,10 +92,90 @@ extern void set_up_apple_events();
 std::unordered_map<std::string, std::shared_ptr <iEventListener>> event_listeners;
 cDrawableManager drawable_mgr;
 
+fs::path scenedit_dir;
+fs::path game_dir;
+fs::path game_binary;
+extern std::string last_load_file;
+
+enum eLaunchType {LOC,START,ENTRANCE};
+
+void launch_scenario(eLaunchType type) {
+	if(boost::ends_with(last_load_file, ".exs")){
+		showError("The scenario editor cannot launch an unpacked scenario directly. You'll need to re-open the scenario from its .boes archive.");
+		return;
+	}
+
+	// Make sure scenario is loaded and currently editing the terrain of a town or outdoor section
+	if(type == LOC){
+		if(overall_mode >= MODE_MAIN_SCREEN){
+			showError("Must be viewing the terrain of a town or outdoor section at the place where you want to put the debug party.");
+			return;
+		}
+	}
+	// Make sure scenario is loaded
+	else if(overall_mode == MODE_INTRO_SCREEN){
+		showError("Must have a scenario loaded.");
+		return;
+	}
+
+	// Prompt to save first
+	if(!save_check("save-before-launch", false)) return;
+
+	fs::path game_binary;
+#ifdef SFML_SYSTEM_MACOS
+	game_binary = "Blades of Exile.app/Contents/MacOS/Blades of Exile";
+#elif defined(SFML_SYSTEM_WINDOWS)
+	game_binary = "Blades of Exile.exe";
+#elif defined(SFML_SYSTEM_LINUX)
+	game_binary = "Blades of Exile";
+#endif
+
+	game_binary = bp::search_path(game_binary, {fs::current_path()});
+	if(game_binary.empty()){
+		showError("The current working directory does not contain Blades of Exile.");
```

Comment:
This can also be a new issue to open after merging, as it seems like a very low priority, but… at this point it would be logical to put up a file selection dialog to allow the user to _locate_ the game binary.